aspnetcore/fundamentals/app-state/samples/3.x/SessionSample/Extensions/SessionExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+
+namespace Web.Extensions
+{
+    #region snippet1
+    public static class SessionExtensions
+    {
+        public static void Set<T>(this ISession session, string key, T value)
+        {
+            session.SetString(key, JsonSerializer.Serialize(value));
+        }
+
+        public static T Get<T>(this ISession session, string key)
+        {
+            var value = session.GetString(key);
+            return value == null ? default : JsonSerializer.Deserialize<T>(value);
+        }
```

Comment:
@javiercn any recommendation regarding this extension method. I'm not certain which one is a better guidance:
```suggestion
        public static bool TryGet<T>(this ISession session, string key, out value)
        {
            var state = session.GetString(key);
            value = default;
            if (state== null)
                return false;
            value = JsonSerializer.Deserialize<T>(state);
            return true;
        }
```

Reply:
FWIW, I like the `TryGet` pattern, but I'd go with the original approach here because:

1. It's a bit simpler for a docs topic.
2. As I understand it, there's no `TryGetString`, so the original approach is consistent with the existing `ISession` API.